contracts/core/Allo.sol

```diff
@@ -464,14 +464,14 @@ contract Allo is
     /// @notice Internal function to check is caller is pool manager
     /// @param _poolId The pool id
     /// @param _address The address to check
-    function _checkOnlyPoolManager(uint256 _poolId, address _address) internal view {
+    function _checkOnlyPoolManager(uint256 _poolId, address _address) internal view virtual {
```

Comment:
Why do we add virtual in all these calls? Is it needed for smock?

Reply:
[Yes!](https://github.com/defi-wonderland/smock-foundry?tab=readme-ov-file#gotchas)